### DISCUSSION TITLE
Jpa 와 Mybatis 관련 사항

### DISCUSSION THREAD

**Discussion by bbbbooo** [goalapa/spring-cacamuca#58] [General]
## command에서 조회를 할때는 query의 조회 메소드를 호출하는게 아닌 command의 jpa를 통해 조회 데이터를 받아야합니다.

현재 스프링 계층이 크게 command(CUD), query(R) 이 두가지로 나뉘어있고 command는 Jpa, query는 mybatis를 사용하고 있어요. 이때, command에서 데이터 조회를 위해 query의 메소드를 사용하면 문제가 발생합니다. query는 mybatis를 사용하는데 command는 Jpa를 사용하고 있어 둘이 바라보는 데이터가 다릅니다. 

*가령,*
1. review의 like_cnt(좋아요 개수)를 +1 하기 위해 query의 조회 메소드를 통해 review를 할당 받는다고 가정하겠습니다.
2. 이때 불러온 review는 mybatis를 통해서 받은 데이터이므로 review.setLikeCnt(review.getLikeCnt() + 1);을 해도 DB의 like_cnt는 변하지 않습니다.
3. 서로 독립된 데이터를 가지고 있기 때문에 실질적인 데이터는 변하지 않게 됩니다. 따라서,  command에서 조회를 할때는 query의 조회 메소드를 호출하는게 아닌 command의 jpa를 통해 조회 데이터를 받아야합니다.

다음은 위와 관련된 코드입니다.

```java
@Transactional
    public void countHeart(Integer no, Integer memberNo) {
        // command의 reviewRepository를 통해 review를 가져옴. (no = 리뷰글 번호)
        Review review = reviewRepository.findById(no).get();
        
        // 좋아요 Entity
        LikeId likeId = new LikeId();

        likeId.setReviewNo(no);
        likeId.setMemberNo(memberNo);

        Like like =  likeRepository.findById(likeId).get();

        review.setLikeCnt(review.getLikeCnt() + 1);
}
```

참고하시어 작업하시면 되겠습니다.

**Comment by myrhymetree** [comment]
확인했습니다

**Comment by Oleevia01** [comment]
확인했습니다!
